addons/website/models/website_page.py

```diff
@@ -209,8 +144,26 @@ def unlink(self):
         return super(Page, self).unlink()
 
     def write(self, vals):
-        if 'url' in vals and not vals['url'].startswith('/'):
-            vals['url'] = '/' + vals['url']
+        for page in self:
+            website_id = page.website_id and page.website_id.id
+            if 'website_id' in vals:
+                website_id = vals['website_id']
```

Comment:
`website_id = page.website_id and page.website_id.id` means that if there is no website, it will return an empty recordset (confusing with the variable name and does it even work well to pass a recordset instead of an id/false as website_id in context?

You can maybe just do smth like that instead:
```py
Website = self.env['website']
if vals.get('website_id) or page.website_id:
    Website = Website.with_context(website_id=vals.get('website_id') or page.website_id.id)
```
And below replace the 2 `self.env['website'].with_context(website_id=website_id)` by just `Website` (will also be easier to read)

Reply:
If you remove the `get_valid_page_url` (see other comment), you will probably have to code it like I suggested, I let you check